compiler/rustc_trait_selection/src/traits/fulfill.rs

```diff
@@ -599,7 +600,16 @@ impl<'a, 'b, 'tcx> FulfillProcessor<'a, 'b, 'tcx> {
                                 unevaluated,
                                 Some(obligation.cause.span),
                             ) {
-                                Ok(val) => Ok(Const::from_value(self.selcx.tcx(), val, c.ty())),
+                                Ok(Some(val)) => {
+                                    Ok(Const::from_value(self.selcx.tcx(), val, c.ty()))
```

Comment:
I think some of this logic could go into a helper method on InferCtxt (that internally calls const_eval_resolve) as it's replicated at least twice